rpctest/rpc_harness.go

```diff
@@ -261,6 +261,17 @@ func (h *Harness) SetUp(createTestChain bool, numMatureOutputs uint32) error {
 // NOTE: This method and SetUp should always be called from the same goroutine
 // as they are not concurrent safe.
 func (h *Harness) TearDown() error {
+	harnessStateMtx.Lock()
+	defer harnessStateMtx.Unlock()
+
+	return h.tearDown()
+}
+
+// tearDown stops the running rpc test instance. See the publicly exported
+// TearDown method for more details.
+//
+// NOTE: This MUST be called with the harnessStateMtx held.
+func (h *Harness) tearDown() error {
```

Comment:
No worries, I removed my commit fixing the race condition as a part of my latest rebase.